tasks/main.yml

```diff
@@ -0,0 +1,99 @@
+---
+- name: Determine JAVA_HOME
+  shell: readlink -f /usr/bin/java | sed "s:bin/java::"
+  register: system_java_home
+  changed_when: false
+
+- name: Set GOCD_COMMON_JAVA_HOME
+  set_fact:
+     gocd_common_java_home: "{{system_java_home.stdout}}"
+
+- name: Thoughtworks go apt repository
+  sudo: yes
+  apt_repository:
+    repo: 'deb {{gocd_common_debian_repository}} /'
+    state: present
+
+# Note: If the user or group exists, but under another ID, Ansible will try and change it.
+- name: Add go group
+  sudo: yes
+  group:
+    name: "{{gocd_common_group}}"
+    state: present
+
+- name: Add go user with home at /var/go
+  sudo: yes
+  user:
+    name: "{{gocd_common_user}}"
+    comment: "Go CD User"
+    group: "{{gocd_common_group}}"
+    home: /var/go
+    shell: /bin/bash
+
+- name: Gos .ssh folder
+  file:
+    path: /var/go/.ssh
+    state: directory
+    group: "{{gocd_common_group}}"
+    owner: "{{gocd_common_user}}"
+  when: gocd_common_configure_ssh
+
+- name: SSH public key
+  copy:
+    dest: /var/go/.ssh/{{GOCD_COMMON_SSH_PUBLIC_KEY | basename}}
+    src: "{{gocd_common_ssh_public_key}}"
+    group: "{{gocd_common_group}}"
+    mode: 0644
+    owner: "{{gocd_common_user}}"
+  when: gocd_common_configure_ssh
+
+- name: SSH private key
+  copy:
+    dest: /var/go/.ssh/{{gocd_common_ssh_private_key | basename}}
+    src: "{{gocd_common_ssh_private_key}}"
+    group: "{{gocd_common_group}}"
+    mode: 0600
+    owner: "{{gocd_common_user}}"
+  when: gocd_common_configure_ssh
```

Comment:
Where is the private key retrieved from ? It should be in the S3 bucket and downloaded using Iam role.